source/common/tcp_proxy/tcp_proxy.h

```diff
@@ -117,7 +118,15 @@ class TunnelingConfigHelperImpl : public TunnelingConfigHelper {
           config_message)
       : hostname_(config_message.hostname()), use_post_(config_message.use_post()),
         header_parser_(Envoy::Router::HeaderParser::configure(config_message.headers_to_add())) {}
-  const std::string& hostname() const override { return hostname_; }
+  const std::string host(const absl::optional<absl::string_view> requested_server_name =
+                             absl::nullopt) const override {
+    auto [hostname, port] =
+        Network::Utility::hostAndPortFromRequestTargetInAuthorityForm(hostname_);
```

Comment:
Does this throw an exception when `hostname_` has no `:`? I think we allowed hosts without full port form before.
Also, could we avoid parsing it every time? Would make more sense to only parse once and reject bad config rather than throwing on data path.

Reply:
Checking on the spec: I think CONNECT requires `:` but it's no longer required in extended CONNECT. I also didn't find any place where it's enforced, so this probably deserves better validation on configuration.

Reply:
Extended CONNECT was added later to h2 to support websockets and other non-TCP payloads: https://datatracker.ietf.org/doc/html/rfc8441.